project/ottersome/README.md

```diff
@@ -0,0 +1,186 @@
+modules like pyplot and numpy. At its core however, it would handle the
+mathematical operations such as convolution, matrix multiplication, element-wise
+multiplication and so on. If time allows, parallelism will be used to speed up
+most of the operations that allow it. 
+
+### Compression
+
+This class or submodule would cover all possible compression algorithms. At the
+moment I only plan to implement jpg compression but the class will be designed
+with expansion to other implementations in mind.
+
+A small diagram can be referenced [here](https://people.cs.rutgers.edu/~elgammal/classes/cs334/slide9_short.pdf)
+
+Though I'll place it here for convenience:
+
+![Block Diagram for JPEG Encoder, All credit goes to Li & Drew](./Images/jpg_compression.png)
+
+
+### ImgSeq
+
+The image sequence class would be responsible for execution of video-like
+feature of the Image Matrix Class. It would deal directly with the Image/Matrix
+class and output to disk the final image in a standardized format such as `.mp4`.
+
+### Kernel 
+
+This class might inherit form the `Matrix` class above.
+Additional features specific to filters might be added, such as :padding, stride and the
+like. 
+
+### Filters
+
+This will just be a collection of predefined kernels that can be easily applied
+to an image.
+
+# API Description
+
+As per the specifications above the user will make use of all the different
+modules to work with images.
+
+```
+
+from racimg import io,mat,matseq,kernel,filters
+
+
+def stich_images():
+    # Some operation could yield numpy matrix here(like a generative deep network)
+    matrices = #Matrix as described above
+    #...
+    for matrix in matrices
+        io.stitcher.add_frame(matrix)
+
+    io.export('./VideoOutput/OutputVideo.mp4',spec='mp4')
+
+def filter_n_compress_images(img_path):
+    # Say we have a raw image
+    img = io.load(img_path)
+
+    # Do some post procesing here
+    kernel = np.array([[-1,-1,-1],[-1,9,-1],[-1,-1,-1]])
+    filters.apply(img,kernel)
+
+    compresed_img io.compress.jpg(matrix)
+    io.export('./Images/compressed_final_img.jpg')
+
+def sobel_application():
+
+
+def main():
+    # Use any of the methods above
+
+```
+
+# Engineering Infrastructure
+
+The project will use CMAKE  so as to make it as cross-platform as possible. 
+Version control will simply be the standard git approach where new features will
+require new branches so as to keep changes atomic. Pull requests 
+of a (potentially non-existent) community will be used for adding features.
+For testing framework I will simply use the google test unit test library with
+continuous integration if possible. 
+For documentation I would like to make it as automatic as possible through
+doxygen, use of Sphynx would be explored if necessary.
+
+# Schedule
+
+For the schedule I will exclude one of the feature in the hopes to have a more
+realistic timeline.
+
+* Week 1 (4/4): Create Math and I/O framework
```

Comment:
I encourage you to set up testing framework and CI along with the I/O framework.  Doing the testing in the second week like you are planning below is fine.  My point is that I/O is something frequently goes wrong and using testing and CI to "clamp" it is significantly helpful to development.